.github/workflows/dotnet-core-build-and-tests.yml

```diff
@@ -0,0 +1,54 @@
+name: .NET Core
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup .NET Core
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 3.1.301
+   
+    - name: Install dependencies for fappdeploy
+      run: |
+           cd ./fappdeploy/
+           dotnet restore
+           
+    - name: Build fappdeploy
+      run: |
+           cd ./fappdeploy/
+           dotnet build --configuration Release --no-restore  
+           
+    - name: Install dependencies for FunctionTestHelper
+      run: |
+           cd ./FunctionTestHelper/
+           dotnet restore
+           
+    - name: Build FunctionTestHelper
+      run: |
+           cd ./FunctionTestHelper/
+           dotnet build --configuration Release --no-restore  
+           
+    - name: Install dependencies for functionAppTests
+      run: |
+           cd ./functionAppTests/
+           dotnet restore
+           
+    - name: Build functionAppTests
+      run: |
+           cd ./functionAppTests/
+           dotnet build --configuration Release --no-restore   
+                 
+    - name: Test functionAppTests
+      run: |
+           cd ./functionAppTests/
+           dotnet test --no-restore --verbosity normal
```

Comment:
you don't need such complex workflow file


just execute two steps from root folder. 

dontnet restore
dotnet test